tls_process_discoverer.go

```diff
@@ -15,14 +14,14 @@ type podInfo struct {
 	cgroupIDs    []uint64
 }
 
-func (t *Tracer) updateTargets(addedWatchedPods []v1.Pod, removedWatchedPods []v1.Pod, addedTargetedPods []v1.Pod, removedTargetedPods []v1.Pod) error {
+func (t *Tracer) updateTargets(addedWatchedPods []api.TargetPod, removedWatchedPods []api.TargetPod, addedTargetedPods []api.TargetPod, removedTargetedPods []api.TargetPod) error {
 	for _, pod := range removedTargetedPods {
 		if t.packetFilter != nil {
 			if err := t.packetFilter.DetachPod(string(pod.UID)); err == nil {
 				log.Info().Str("pod", pod.Name).Msg("Detached pod from cgroup:")
 			}
 		}
-		wInfo, ok := t.watchingPods[pod.UID]
+		wInfo, ok := t.watchingPods[types.UID(pod.UID)]
```

Comment:
It's better to refactor `watchingPods` to `map[string]*watchingPodsInfo` to avoid type casting